cylc/flow/graph_parser.py

```diff
@@ -933,8 +941,13 @@ def _compute_triggers(
                     ) from None
             else:
                 fam = False
-                if not output:
-                    # Make implicit success explicit.
+                # RHS: don't infer ":succceeded", except:
```

Comment:
Technically this is a slightly different expression, but I've put some thought into accessing the `else:` statement by trying to create a situation where `not output & not optional & expr`, and have been unable to do it.

Reply:
(no longer relevant)